### PR TITLE
feat: Adding support for profile tokens

### PR DESCRIPTION
**Pull request by jdalrymple**
Request Profile Tokens https://docs.gitlab.com/ee/administration/monitoring/performance/request_profiling.html

closes: #441

### CODE REVIEW

File: src/core/infrastructure/BaseService.ts [hunk 1]
```diff
@@ -34,6 +36,10 @@ export class BaseService {
     else if (jobToken) this.headers['job-token'] = jobToken;
     else if (token) this.headers['private-token'] = token;
 
+    // Profiling
+    if (profileToken) this.headers['X-Profile-Token'] = profileToken;
+    if (profileMode) this.headers['X-Profile-Mode'] = profileMode;
```
Comment (casz) [inline src/core/infrastructure/BaseService.ts]: nit: this will always be added, it should only be added if `x-profile-token` is set

Comment (jdalrymple) [reply src/core/infrastructure/BaseService.ts]: 🤦‍♂ yup, overlooked that.

Comment (jdalrymple) [reply src/core/infrastructure/BaseService.ts]: Fixed

**Comment by casz** [comment]
CI failed 😓

**Comment by jdalrymple** [comment]
Yea, I can't fix the linting issues on my current pc :( It will have to wait until I get home for me to fix it.

**Comment by casz** [comment]
Gogo docker? 
```
docker run --rm -it -w /src -v ${PWD}:/src node
npm i
npm run lint:fix
npm run lint:doc:fix
```

**Comment by jdalrymple** [comment]
Also don't have docker locally lol :P

**Comment by casz** [comment]
Why wouldn't you 😆

**Pull request merged**